File: examples/04_application/stock/stock.php
<?php
// Lean Code — BSD 3-Clause License — Vivian Voss, 2026
// Scope: Read and check stock availability.

namespace Stock;

function readStockForProduct(string $product, string $stockPath): int
{
    if (!file_exists($stockPath)) {
        return 0;
    }

    foreach (explode("\n", trim(file_get_contents($stockPath))) as $line) {
        [$name, $count] = explode(',', $line, 2);
        if ($name === $product) {
            return (int) $count;
        }
    }

    return 0;
}

function checkStockAvailability(string $product, int $quantity, string $stockPath): bool
{
    $stock = readStockForProduct($product, $stockPath);
    return $stock >= $quantity;
}

File: examples/04_application/stock/stock.go
// Lean Code — BSD 3-Clause License — Vivian Voss, 2026
// Scope: Read and check stock availability.

package stock

import (
	"os"
	"strconv"
	"strings"
)

func ReadStockForProduct(product string, stockPath string) int {
	data, err := os.ReadFile(stockPath)
	if err != nil {
		return 0
	}

	for _, line := range strings.Split(strings.TrimSpace(string(data)), "\n") {
		parts := strings.SplitN(line, ",", 2)
		if parts[0] == product {
			count, _ := strconv.Atoi(parts[1])
			return count
		}
	}

	return 0
}

func CheckStockAvailability(product string, quantity int, stockPath string) bool {
	s := ReadStockForProduct(product, stockPath)
	return s >= quantity
}
